### DISCUSSION TITLE
Apply BC only to nodes in contact

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#20931] [Q&A General] [answered]
Hello,

I'd like to apply a BC only to nodes that are currently in contact. Is there a way to do this in MOOSE?

**Comment by GiudGiud** [comment]
So boundary conditions on nodes that are in between two blocks? Internal to the mesh?

**Reply by Flolaffel** [reply]
BCs only on nodes from one block that are currently in contact. See my description for @dschwen

**Comment by dschwen** [comment, marked as answer]
Can you elaborate on what physics you are trying to model? We have something like this in mortar based pressure dependent gap heat conduction.

**Reply by Flolaffel** [reply]
Of course. I'm doing a tensor mechanics, contact, porous flow simulation. My goal is to indent a porous medium from the top so fluid is squeezed out. At the same time the top surface is the only unconfined surface, so I'm applying a BC there so the porepressure is 0 and there can be outflow. But I want to restrict the outflow at the contact area because fluid wouldn't be able to flow out where contact is established. So I was wondering if MOOSE defines a nodeset that contains the nodes in contact which I could to use to prohibit outflow on these specific nodes.

**Reply by recuero** [reply]
You can define a nodeset or sideset and then probably add a tailored mortar constraint that would do what you want. It doesn't sound like your variable is temperature, so the object may look slightly different from what we already have in MOOSE.

**Reply by Flolaffel** [reply]
The variable would be the porepressure and it should be 0 on the free surface but unrestricted where contact is established. I just noticed that would mean I'd have to set my BC only on the nodes NOT in contact, not like I formulated my question initially.

I will look into the mortar heat conductance.
